doc/reference/reference_lua/checks.rst

```diff
@@ -0,0 +1,516 @@
+Otherwise, an error is raised.
+
+..  literalinclude:: /code_snippets/test/checks/checks_string_test.lua
+    :language: lua
+    :end-before: -- Tests
+    :dedent:
+
+
+.. _checks_multiple_arguments:
+
+Multiple arguments
+~~~~~~~~~~~~~~~~~~
+
+To check the types of several arguments, you need to pass the corresponding type qualifiers to the ``checks`` function.
+In the example below, both arguments should be string values.
+
+..  literalinclude:: /code_snippets/test/checks/checks_string_multiple_test.lua
+    :language: lua
+    :end-before: -- Tests
+    :dedent:
+
+To turn off checking of specific arguments, use the :ref:`? placeholder <checks_any_type>`.
+
+
+.. _checks_variable_number_of_arguments:
+
+Variable number of arguments
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+You can check the types of explicitly specified arguments for functions that accept a variable number of arguments.
+
+..  literalinclude:: /code_snippets/test/checks/checks_vararg_test.lua
+    :language: lua
+    :end-before: -- Tests
+    :dedent:
+
+
+
+.. _checks_string_type_qualifier:
+
+String type qualifier
+---------------------
+
+This section describes how to check a specific argument type using a string type qualifier:
+
+- The :ref:`Supported types <checks_string_supported_types>` section describes all the types supported by the ``checks`` module.
+- If required, you can make a :ref:`union type <checks_union_type>` to allow an argument to accept several types.
+- You can make any of the supported types :ref:`optional <checks_optional_type>`.
+- To turn off checking of specific arguments, use the :ref:`? placeholder <checks_any_type>`.
+
+.. _checks_string_supported_types:
+
+Supported types
+~~~~~~~~~~~~~~~
+
+.. _checks_lua_types:
+
+Lua types
+*********
+
+A string type qualifier can accept any of the `Lua types <https://www.lua.org/pil/2.html>`_, for example, ``string``, ``number``, ``table``, or ``nil``.
+In the example below, the ``checks`` function accepts ``string`` to validate that only a string value can be passed to the ``greet`` function.
+
+..  literalinclude:: /code_snippets/test/checks/checks_string_test.lua
+    :language: lua
+    :end-before: -- Tests
+    :dedent:
+
+
+.. _checks_tarantool_types:
+
+Tarantool types
+***************
+
+You can use :ref:`Tarantool-specific types <index_box_field_type_details>` in a string qualifier.
+The example below shows how to check that a function argument is a decimal value.
+
+..  literalinclude:: /code_snippets/test/checks/checks_decimal_test.lua
+    :language: lua
+    :end-before: -- Tests
+    :dedent:
+
+This table lists all the checks available for Tarantool types:
+
+..  container:: table
+
+    ..  list-table::
+        :widths: 30 40 30
+        :header-rows: 1
+
+        *   -   Check
+            -   Description
+            -   See also
+
+        *   -   ``checks('datetime')``
+            -   Check whether the specified value is :ref:`datetime_object <datetime_obj>`
```

Comment:
I don't see much point in repeating the same sentence in all rows changing only the type (which is obvious from the first column). Can we restructure the table somehow?

Reply:
Auto-generated lists of  API members often look like this. I think, it's not a problem.